packages/smart-account/src/SmartAccount.ts

```diff
@@ -307,24 +315,41 @@ class SmartAccount extends EventEmitter {
   // Optional methods for connecting paymaster
   // Optional methods for connecting another bundler
 
-  public async sendGasLessTransaction(
+  public async sendGaslessTransaction(
     transactionDto: TransactionDto
   ): Promise<TransactionResponse> {
     let { version, chainId } = transactionDto
     const { transaction } = transactionDto
     chainId = chainId ? chainId : this.#smartAccountConfig.activeNetworkId
     version = version ? version : this.DEFAULT_VERSION
-    const aaSigner = this.aaProvider[this.#smartAccountConfig.activeNetworkId].getSigner()
+    const aaSigner = this.aaProvider[chainId].getSigner()
 
     await this.initializeContractsAtChain(chainId)
     const multiSendContract = this.contractUtils.multiSendContract[chainId][version].getContract()
 
     const isDelegate = transaction.to === multiSendContract.address ? true : false
 
-    const response = await aaSigner.sendTransaction(transaction, false, isDelegate, this)
+    let customData: any = {}
+    let transactionRequest : TransactionRequest = {...transaction}
 
+    const isDeployed = await this.contractUtils.isDeployed(
+      chainId,
+      this.DEFAULT_VERSION,
+      this.address
+    )
+
+    if(!isDeployed || isDelegate) {
+       customData.appliedGasLimit = ethers.constants.Two.pow(24) // estimateGas for execFromEntryPoint
```

Comment:
Why hardcoded value here?

Reply:
so userOp callGasLimit is estimation on execFromEntryPoint method on wallet.

```
// Review: doesn't work if wallet is not already deployed.
    const callGasLimit =
      parseNumber(detailsForUserOp.gasLimit) ??
      (await this.provider.estimateGas({
        from: this.entryPoint.address,
        to: this.getWalletAddress(),
        data: callData
      }))
```
      
 so if wallet is not deployed it just estimates and returns 21000 + data cost  then it fails
 
 Further changes is marked for testing / brainstorming and to be launched in next release before gasless goes live on 
 mainnet.